### PR TITLE
Add travis-ci.org testing support

### PR DESCRIPTION
**Pull request by AmeBel**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# TODO: one day when container infrastructure works better change to
+# sudo: false
+sudo: required
+
+cache: apt
+
+before_install:
+    - sudo apt-add-repository -y ppa:opencog-dev/ppa
+    - sudo apt-add-repository -y ppa:boost-latest/ppa
+    - sudo apt-add-repository -y ppa:ubuntu-toolchain-r/test
```
Comment (linas) [inline .travis.yml]: why all these ppa's? I run all of opencog without pulling any ppa's ...

Comment (AmeBel) [inline .travis.yml]: travis builds run on ubuntu 12.04, thus the ppas.

Comment (linas) [inline .travis.yml]: ah, OK.  Could you add a note/comment saying this?

**Comment by linas** [comment]
well, I'm happy to merge this, except that a bunch of unit tests fail. which I guess means that it should not be merged ..!?

**Comment by AmeBel** [comment]
The failures could be b/c of a non complete build,  as a result of have multiple jobs
https://travis-ci.org/opencog/moses/builds/73374811#L1290-L1294 

Are there some make flags that could be used to avoid such failures while having multiple builds? Would the load-average flag help? 

VM detail per build  job on travis =  1.5 cores and 3GB of memory

**Comment by AmeBel** [comment]
Tried different configuration but no chages in the list of test that are SEGFAULTing.

https://travis-ci.org/AmeBel/moses/builds/73489548#L1731-L1740
https://travis-ci.org/AmeBel/moses/builds/73500956#L1735-L1745

**Comment by ngeiswei** [comment]
Could these failures be related to https://github.com/opencog/cogutils/pull/9 ?

**Comment by AmeBel** [comment]
tested on buildbot http://buildbot.opencog.org:8010/builders/moses_master_trusty/builds/3/steps/test/logs/stdio all good

**Pull request closed without merging**